File: 백준/Silver/25916. 싫은데요/싫은데요.py
n, m = map(int, input().split())
a = list(map(int, input().split()))

left, right = 0, 0
mx, v = 0, 0;
while right < n:
    v += a[right]
    right += 1

    while v > m:
        v -= a[left]
        left += 1

    mx = max(mx, v)

print(mx)

File: 백준/Silver/25916. 싫은데요/싫은데요.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;


public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        int[] a = Arrays.stream(br.readLine().split(" "))
                .mapToInt(Integer::parseInt)
                .toArray();

        int left = 0, right = 0;
        int mx = 0, v = 0;
        while (right < n) {
            v += a[right++];

            while (v > m)
                v -= a[left++];

            mx = Math.max(mx, v);
        }

        System.out.println(mx);
    }
}